### DISCUSSION TITLE
Adding Favorites and Playback services to Media Source

### DISCUSSION THREAD

**Discussion by giachello** [home-assistant/architecture#770] [General]
Hi
So @frenck  and I were discussing this [PR #72730](https://github.com/home-assistant/core/pull/72730) where I want to add a "Favorites" function to Radio Browser, and services to start playback (to allow for automations).

He suggested that the favorites function should sit with Media Source instead since it's not limited to radios, and that we discuss the proposal here. 

I am thinking that we could add either a config panel or a UI feature to select favorites in the media panel, and then a way to start a media source playback.

@hunterjm  et al what is your take? What would be useful for a productive discussion of this topic?

**Comment by balloob** [comment]
I guess the challenge is on what level we would want to implement favorites:

- media source level. It will show up as a new folder for every media player that supports media source. Favorites will be limited to media sources.
- media player level. It will show up just for that media player but any item offered via the media browser for that media player can be favorited.

**Reply by giachello** [reply]
I  think media source should be the place for favorites.

I feel like the media player use case (having favorites for any content) is already solved by the app sending the content to media player e.g., Spotify already has a favorites function, Apple Music does too. ...

IMO the player is just  an interface to the output and it shouldn't have an opinion of what is favorite. 

There are also 2 positive reasons for using media sources for this:

1. I think of media sources as a way to establish a content library domestic to HA , just like an external music app like Spotify has a library of content. The library should have an opinion of what a favorite asset is, not the player. 
2. In addition, favorites are such independently of where I decide to play them back.

**Reply by giachello** [reply]
@balloob  any thoughts on this? would love your opinion so we can make progress. I'll have some time in the next few weekends to look into it.

**Comment by Ed1123** [comment]
I know is over 3 years after the initial issue was opened, but did it ever got implemented or there is something planned?
